Assets/Scripts/Simulation/Simulator.cs

```diff
@@ -472,6 +473,27 @@ static void ProcessBuiltinChip(SimChip chip)
 					chip.OutputPins[1].State = (ushort)(data & ByteMask);
 					break;
 				}
+				case ChipType.Port_In:
+				{
+					int? portIndex = PortRegistry.GetPortIndex(chip);
+					if (portIndex == null) throw new Exception("QLKIjdsflajl");
+
+					// Read from external source
+					byte portValue = PortCommunicationManager.ReadInputPort(portIndex.Value);
+					PinState.Set(ref chip.OutputPins[0].State, portValue);
+					break;
+				}
+				case ChipType.Port_Out:
+				{
+					int? portIndex = PortRegistry.GetPortIndex(chip);
+					if (portIndex == null) break;
```

Comment:
Consistency: If you throw an exception for one case, be sure to throw it for similar situations too; or in neither.